lib/src/steps/multi_step_step.dart

```diff
@@ -0,0 +1,110 @@
+import 'package:automated_testing_framework/automated_testing_framework.dart';
+import 'package:logging/logging.dart';
+import 'package:meta/meta.dart';
+
+/// Test step that groups different [TestStep] to be executed
+/// as a block.
+class MultiStepStep extends TestRunnerStep {
+  MultiStepStep({
+    this.name,
+    @required this.steps,
+  }) : assert(steps != null);
+
+  final String name;
+  final List<dynamic> steps;
+
+  /// Creates an instance from a JSON-like map structure.  This expects the
+  /// following format:
+  ///
+  /// ```json
+  /// {
+  ///   "name": <String>,
+  ///   "steps": <List>
+  /// }
+  /// ```
+  static MultiStepStep fromDynamic(dynamic map) {
+    MultiStepStep result;
+
+    if (map != null) {
+      var name = map['name'] ?? 'Default MultiStep';
+      List stepsList = map['steps'] is List ? map['steps'] : [map['steps']];
+
+      result = MultiStepStep(
+        name: name,
+        steps: stepsList
+          ..removeWhere(
+            (step) => step == null,
+          ),
+      );
+    }
+
+    return result;
+  }
+
+  /// Executes the step. This will iterate through the List of steps
+  /// and will await the execution of each one.
+  @override
+  Future<void> execute({
+    TestReport report,
+    TestController tester,
+  }) async {
+    log(
+      'MultiStep: Starting execution of $name',
```

Comment:
Please change all `MultiStep` to `multi_step` so the logs match the id of the step.  This provides consistency in the logs, and also makes it easier to tie the logs to the test JSON.